YAFCmodel/Serialization/SerializationMap.cs

```diff
@@ -9,6 +9,9 @@ namespace YAFC.Model
     [AttributeUsage(AttributeTargets.Property)]
     public class SkipSerializationAttribute : Attribute {}
 
+    [AttributeUsage(AttributeTargets.Property)]
+    public class NoUndoAttribute : Attribute {}
```

Comment:
is there a reason for having this property besides documentation of intended behavior?

if not and the other properties are the same then that's fine, but if not and the other properties actually influence the behavior by themselves I'd like you to think more closely whether the attribute is the right choice

Reply:
Yes, it's needed this way. And I prefer to name things so it's instantly clear why it's used.

I tried and when i use [SkipSerialization] it's also not saving the expanded state instead of just ignoring it for undo.

What alternative solution so you suggest?

Reply:
the thing that tripped me off is that the existence of `NoUndoAttribute` sets type to `PropertyType.NoUndo` which has no further reference in the code. it's only important to switch away from PropertyType.Normal because that check is used in `SpecificSerializationMap.BuildUndo`/`ReadUndo`

you could make it more obvious in BuildUndo/ReadUndo by blacklisting the three other PropertyType values instead of whitelisting PropertyType.Normal, but I'm not even sure about that.

so forget about my comment. I mainly wanted to make sure that this attribute is really needed in some way and not a leftover of a previous approach that didn't work out in the end.